### ISSUE TITLE
Inconsistent parsing for non-nullable params

### ISSUE THREAD

**Issue by m-ronchi** [adelsz/pgtyped#372]
Hi,
I noticed that inferring types from parameters is inconsistent.

for example:
this
```
/* @name ExampleQuery */
UPDATE "Foo"
SET "foo" = :foo!,
    "bar" = null
WHERE "externalId" = :id!;
```
produces (wrong type):
```
/** 'ExampleQuery' parameters type */
export interface IExampleQueryParams {
  foo: string | null | void;
  id: string;
}
```

while this (notice a single space difference)
```
/* @name ExampleQuery */
UPDATE "Foo"
SET "foo" = :foo! ,
    "bar" = null
WHERE "externalId" = :id!;
```
produces (correct type):
```
/** 'ExampleQuery' parameters type */
export interface IExampleQueryParams {
  foo: number;
  id: string;
}
```

It appears that pgtyped can't parse the `!,` tokens without whitespace in between...?

**Comment by adelsz** [comment]
This is definitely a bug in our parser, thanks for reporting.

**Comment by atmiguel** [comment]
I just encountered this same issue. Just to add another case where the non-nullable params don't parse correctly:
```
SELECT * from "Foo"
WHERE ("foo" = :foo! OR "bar" = :bar!) AND "baz" = :baz!;
```

The query generation encounters the following error:
```
Error in query. Details: {
  errorCode: 'op_error',
  hint: 'No operator matches the given name and argument types. You might need to add explicit type casts.',
  message: 'operator does not exist: text = numeric',
  position: '34'
}
```

And the solution is the same, adding a space between `:bar!` and `)` fixes the issue.

**Comment by adelsz** [comment]
Fixed in v0.13.0
